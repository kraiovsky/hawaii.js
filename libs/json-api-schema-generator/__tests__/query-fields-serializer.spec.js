const serializeFields = require('../lib/query-fields-serializer')
const testCases = require('./query-fields-serializer.cases')

let fieldsArr

describe('serializeFields()', () => {
  describe.each(testCases.serializeFields.happy)('', (fields, expected) => {
    beforeEach(() => {
      fieldsArr = serializeFields(fields)
    })

    test(`👍 should return ${expected}`, () => {
      expect(fieldsArr).toMatchSnapshot()
    })
  })
})
