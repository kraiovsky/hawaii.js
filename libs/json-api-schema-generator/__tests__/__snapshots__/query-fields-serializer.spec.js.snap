// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`serializeFields()  👍 should return array of field names passed in query 1`] = `
Array [
  "name",
  "test",
]
`;

exports[`serializeFields()  👍 should return array of field names passed in query, clearing empty elements 1`] = `
Array [
  "name",
  "test",
]
`;

exports[`serializeFields()  👍 should return array with one element if only one field name passed 1`] = `
Array [
  "name",
]
`;

exports[`serializeFields()  👍 should return empty array if empty field passed in query 1`] = `Array []`;

exports[`serializeFields()  👍 should return empty array if no field passed in query 1`] = `Array []`;
